adapters/http-base/src/main/java/org/eclipse/hono/adapter/http/impl/VertxBasedHttpProtocolAdapter.java

```diff
@@ -271,9 +286,15 @@ private void addCommandResponseRoutes(final Router router, final Handler<Routing
         // route for uploading command response message
         router.put(commandResponseMatchAllPath)
                 .handler(this::handlePutCommandResponse)
+                .handler(getBodyHandler())
                 .setName(commandResponsePutPathWithParams);
```

Comment:
The order seems wrong here, putting the bodyHandler after the `handlePutCommandResponse` handler.

In general it should also be checked that the bodyHandler isn't unnecessarily added multiple times to routes. Above, an authenticated request would be handled by routes containing the bodyHandler 3 times.